test/jdk/java/awt/print/PrinterJob/XparColor.java

```diff
@@ -75,184 +72,289 @@ public static void main(String s[]) {
         g2.dispose();
-
         return Printable.PAGE_EXISTS;
     }
 
     public void drawDemo(int w, int h, Graphics2D g2) {
-
-        Color reds[] = { Color.red.darker(), Color.red };
+        Color reds[] = {Color.red.darker(), Color.red};
         for (int N = 0; N < 18; N++) {
             float i = (N + 2) / 2.0f;
-            float x = (float) (5+i*(w/2/10));
-            float y = (float) (5+i*(h/2/10));
-            float ew = (w-10)-(i*w/10);
-            float eh = (h-10)-(i*h/10);
+            float x = (float) (5 + i * (w / 2 / 10));
+            float y = (float) (5 + i * (h / 2 / 10));
+            float ew = (w - 10) - (i * w / 10);
+            float eh = (h - 10) - (i * h / 10);
             float alpha = (N == 0) ? 0.1f : 1.0f / (19.0f - N);
-            if ( N >= 16 )
-                g2.setColor(reds[N-16]);
-            else
+            if (N >= 16) {
+                g2.setColor(reds[N - 16]);
+            } else {
                 g2.setColor(new Color(0f, 0f, 0f, alpha));
-            g2.fill(new Ellipse2D.Float(x,y,ew,eh));
+            }
+            g2.fill(new Ellipse2D.Float(x, y, ew, eh));
+        }
+    }
+
+
+    /*****************************************************
+     Standard Test Machinery Section
+     DO NOT modify anything in this section -- it's a
+     standard chunk of code which has all of the
+     synchronisation necessary for the test harness.
+     By keeping it the same in all tests, it is easier
+     to read and understand someone else's test, as
+     well as insuring that all tests behave correctly
+     with the test harness.
+     There is a section following this for test-defined
+     classes
+     ******************************************************/
+    private static boolean theTestPassed = false;
+    private static boolean testGeneratedInterrupt = false;
+    private static String failureMessage = "";
+
+    private static Thread mainThread = null;
+
+    private static int sleepTime = 300000;
+
+    public static void main(String args[]) throws InterruptedException {
+        mainThread = Thread.currentThread();
+        try {
+            init();
+        } catch (TestPassedException e) {
+            //The test passed, so just return from main and harness will interpret this return as a pass
+            return;
+        }
+
+        /*
+            At this point, neither test passed nor test failed has been
+            called -- either would have thrown an exception and ended the
+            test, so we know we have multiple threads.
+            Test involves other threads, so sleep and wait for them to
+            called pass() or fail()
+         */
+        try {
+            Thread.sleep(sleepTime);
+            //Timed out, so fail the test
+            throw new RuntimeException("Timed out after " + sleepTime / 1000 + " seconds");
+        } catch (InterruptedException e) {
+            if (!testGeneratedInterrupt) throw e;
```

Comment:
I guess timeout code is there here but for somehow it is not working as expected..Could you check?
Otherwise there are other ways of handling user interaction done in
java/awt/print/PrinterJob/BannerTest.java
java/awt/print/PrinterJob/TestSaveFileWithoutPrinter.java